### DISCUSSION TITLE
soot for the masses

### DISCUSSION THREAD

**Discussion by PowerStat** [soot-oss/soot#1541] [Ideas]
Hello, maybe I am a bit naive, but what I understand is that soot is a framework for improving/analyzing java code?
If so, I must say as an open source developer - everything I read about soot is interesting but much to complex for me, because I don't want to go into that deep.

What I expected from soot is to have a simple maven plugin like spotbugs/findbugs, pmd, checkstyle that I can use in my projects and improve my code quality. On the other hand it looks like there are nice things like call-graph generator - that would be nice to have in the maven report (mvn site).

So having such a maven-soot-plugin would make soot available to the masses of open source developers.

**Comment by patricklam** [comment]
Soot is generally not designed to be an end user tool but rather a
transformation toolkit. It is possible to build end user tools on top of
Soot but they really are their own thing.
